specification/compute/resource-manager/Microsoft.Compute/DiagnosticRP/preview/2024-03-01-preview/diagnostic.json

```diff
@@ -0,0 +1,665 @@
+          },
+          {
+            "name": "storageConfigurationInput",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/StorageConfigurationInput"
+            },
+            "description": "Storage Configuration object supplied in the body of the Post to cache storageAccount for a given subscription."
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Succeeded"
+          },
+          "default": {
+            "description": "Error response describing why the register operation failed.",
+            "schema": {
+              "$ref": "../../../../../../common-types/resource-management/v5/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Create a request for registering a storageAccount information.": {
+            "$ref": "./examples/RegisterStorageConfiguration.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/diagnostics/diskInspection/validateStorageConfiguration": {
+      "post": {
+        "tags": [
+          "validateStorageConfiguration"
+        ],
+        "operationId": "DiskInspectionStorageConfiguration_Validate",
+        "x-ms-long-running-operation": false,
+        "description": "Validate if a storageAccount configuration exists for a subscription used for DiskInspection.",
+        "consumes": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "../../../../../../common-types/resource-management/v5/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../../common-types/resource-management/v5/types.json#/parameters/LocationParameter"
+          },
+          {
+            "$ref": "../../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/StorageConfigurationResponse"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the get call failed.",
+            "schema": {
+              "$ref": "../../../../../../common-types/resource-management/v5/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Returns storageAccount value for an existing configuration entry": {
+            "$ref": "./examples/ValidateStorageConfiguration.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/diagnostics/spotPlacementRecommender/generate": {
```

Comment:
>generate [](http://example.com/codeflow?start=123&length=8)

why do the diagnostics have different action verbs from each other?

Reply:
Resolving due to earlier discussion

Reply:
I'm not sure we really answered this particular question about the new api.